rotkehlchen/chain/evm/decoding/utils.py

```diff
@@ -188,3 +196,50 @@ def update_cached_vaults(
                 f'Failed to store token information for {display_name} vault '
                 f'due to {error}. Vault: {vault}. Skipping...',
             )
+
+
+def get_vault_price(
+        inquirer: 'Inquirer',
+        vault_token: 'EvmToken',
+        evm_inquirer: 'EvmNodeInquirer',
+        display_name: str,
+        vault_abi: ABI,
+        pps_method: str,
+        pps_method_args: list | None = None,
+) -> Price:
+    """Gets vault token price by multiplying price per share by the underlying token's USD price.
+    Price per share is retrieved from the vault contract using the specified pps_method.
+    Returns the vault token price or ZERO_PRICE on error.
+    """
+    try:
+        price_per_share = evm_inquirer.call_contract(
+            contract_address=vault_token.evm_address,
+            abi=vault_abi,
+            method_name=pps_method,
+            arguments=pps_method_args,
+        )
+    except RemoteError as e:
+        log.error(
+            f'Failed to get price per share for {display_name} '
+            f'vault {vault_token} on {evm_inquirer.chain_id}: {e}',
```

Comment:
Please remember to use chain name for all logging and display purposes.

```suggestion
            f'vault {vault_token} on {evm_inquirer.chain_name}: {e}',
```